samza-core/src/main/java/org/apache/samza/execution/JobGraph.java

```diff
@@ -45,7 +45,7 @@
  * Note that intermediate streams are both the input and output of a JobNode in JobGraph.
  * So the graph may have cycles and it's not a DAG.
  */
-public class JobGraph {
+/* package private */ class JobGraph {
```

Comment:
fwiw, I like it. It assures future readers that the qualifier wasn't forgotten and that it was package private by design.